candig/server/frontend.py

```diff
@@ -864,63 +872,8 @@ def handleException(exception):
                                 mimetype=return_mimetype)
 
 
-def requires_session(f):
-    """
-    Decorator for browser session routes. Inspects tokens and ensures client sessions + server
-    sessions are aligned.
-    """
-
-    @functools.wraps(f)
-    def decorated(*args, **kargs):
-        if app.config.get("TYK_ENABLED"):
-
-            # TODO: CSRF check / checksum
-            if not flask.request.cookies.get("session_id"):
-                flask.session.clear()
-                return flask.redirect(_generate_login_url())
-
-        return f(*args, **kargs)
-    return decorated
-
-
-def startLogin():
-    """
-    If we are not logged in, this generates the redirect URL to the OIDC
-    provider and returns the redirect response
-    :return: A redirect response to the OIDC provider
-    """
-    flask.session["state"] = oic.rndstr(SECRET_KEY_LENGTH)
-    flask.session["nonce"] = oic.rndstr(SECRET_KEY_LENGTH)
-    args = {
-        "client_id": app.oidcClient.client_id,
-        "response_type": "code",
-        "scope": ["openid", "profile"],
-        "nonce": flask.session["nonce"],
-        "redirect_uri": app.oidcClient.redirect_uris[0],
-        "state": flask.session["state"]
-    }
-
-    result = app.oidcClient.do_authorization_request(
-        request_args=args, state=flask.session["state"])
-    return flask.redirect(result.url)
-
-
 @app.before_request
-def checkAuthentication():
-    """
-    The request will have a parameter 'key' if it came from the command line
-    client, or have a session key of 'key' if it's the browser.
-    If the token is not found, start the login process.
-
-    If there is no oidcClient, we are running naked and we don't check.
-    If we're being redirected to the oidcCallback we don't check.
-
-    :returns None if all is ok (and the request handler continues as usual).
-    Otherwise if the key was in the session (therefore we're in a browser)
-    then startLogin() will redirect to the OIDC provider. If the key was in
-    the request arguments, we're using the command line and just raise an
-    exception.
-    """
+def checkAuthorization():
```

Comment:
Why was this function renamed from Authentication to Authorization? I believe what we are doing is _still_ authentication.

Reply:
Because this particular function is running the check for the access_list

Reply:
![image](https://user-images.githubusercontent.com/18109953/72759452-a354f100-3b8a-11ea-92f9-7368ec93441e.png)

It checks the access_list when tyk is enabled

Reply:
oh good to know